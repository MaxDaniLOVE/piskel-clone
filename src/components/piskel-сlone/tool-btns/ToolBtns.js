import React from 'react';

const ToolBtns = ({ activeTool, onToolChange }) => {
  const toolButtons = [
    { tool: 'pencil', icon: 'fa fa-pencil-alt' },
    { tool: 'color-picker', icon: 'fa fa-eye-dropper' },
    { tool: 'bucket', icon: 'fa fa-fill-drip' },
    { tool: 'eraser', icon: 'fa fa-eraser' },
    { tool: 'stroke', icon: 'fa fa-pencil-ruler' },
  ];
  const btns = toolButtons.map(({ tool, icon }) => {
    const isActive = activeTool === tool;
    const newClassName = isActive ? 'btn btn-success tool' : 'btn btn-secondary tool';
    return (
      <button
        type="button"
        className={newClassName}
        key={tool}
        onClick={() => { onToolChange(tool); }}
        data-toggle="tooltip"
        data-placement="right"
        title={tool}
      >
        <i className={icon} aria-hidden="true" />
      </button>
    );
  });
  return btns;
};

export default ToolBtns;
