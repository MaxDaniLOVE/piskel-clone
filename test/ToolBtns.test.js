import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ToolBtns from '../src/components/piskel-сlone/tool-btns/ToolBtns';

configure({adapter: new Adapter()});
describe('ToolBtns', () => {
  const output = shallow(<ToolBtns />);
  it('should render correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should display 5 buttons', () => {
    expect(output.length).toBe(5);
  });
});