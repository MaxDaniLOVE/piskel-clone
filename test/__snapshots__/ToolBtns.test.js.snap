// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolBtns should render correctly 1`] = `
Array [
  <button
    className="btn btn-secondary tool"
    data-placement="right"
    data-toggle="tooltip"
    key="pencil"
    onClick={[Function]}
    title="pencil"
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-pencil-alt"
    />
  </button>,
  <button
    className="btn btn-secondary tool"
    data-placement="right"
    data-toggle="tooltip"
    key="color-picker"
    onClick={[Function]}
    title="color-picker"
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-eye-dropper"
    />
  </button>,
  <button
    className="btn btn-secondary tool"
    data-placement="right"
    data-toggle="tooltip"
    key="bucket"
    onClick={[Function]}
    title="bucket"
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-fill-drip"
    />
  </button>,
  <button
    className="btn btn-secondary tool"
    data-placement="right"
    data-toggle="tooltip"
    key="eraser"
    onClick={[Function]}
    title="eraser"
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-eraser"
    />
  </button>,
  <button
    className="btn btn-secondary tool"
    data-placement="right"
    data-toggle="tooltip"
    key="stroke"
    onClick={[Function]}
    title="stroke"
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-pencil-ruler"
    />
  </button>,
]
`;
